modules/local/locidex_select.nf

```diff
@@ -0,0 +1,194 @@
+        if(db_found){
+            def selected_db = select_locidex_db_path(allele_db_data[db[db_key]], db[db_key])
+            /// Write selected allele database info to a file for the final report
+            write_config_data(selected_db, output_config)
+            scheme = join_database_paths(selected_db)
+            paired_p = db_found
+            break
+        }
+    }
+
+    if(!paired_p){
+        write_config_data(["No database selected."], output_config)
+    }
+
+}
+
+
+def write_config_data(db_data, output_name){
+    /// Config data for db to use
+    def json_data = new JsonBuilder(db_data).toPrettyString()
+    def output_file = file(output_name).newWriter()
+    output_file.write(json_data)
+    output_file.close()
+}
+
+def join_database_paths(db_path){
+    /// Database paths are relative to the manifest, hopefully this will not offer many issue on cloud executors
+    def input_dir_path = [params.lx_allele_database, db_path[params.locidex.manifest_db_path]].join(File.separator)
+    return input_dir_path
+}
+
+def select_locidex_db_path(db_values, db_name){
+    /// Select the optimal locidex database by parsing date fields for the organism
+    /// Database fields are labeled by date, so the most recent will be shown
+    /// Db value is an object containing the path fields and the config fields
+    /// db_values: is the list of database config information in the manifest
+
+
+    def database_entries = db_values.size()
+    def default_date = new SimpleDateFormat(params.locidex.date_format_string).parse("0001-01-01")
+    def max_date = default_date
+    def max_date_idx = 0
+    def idx = 0
+    def dates = []
+
+    // Validate all input fields
+    for(value in db_values){
+        if(!value.containsKey(params.locidex.manifest_db_path)){
+            exit 1, "Missing path value in locidex config for: ${db_name}"
+        }
+        if(!value.containsKey(params.locidex.manifest_config_key)){
+            exit 1, "Missing config data for locidex database entry: ${db_name}"
+        }
+        if(!value[params.locidex.manifest_config_key].containsKey(params.locidex.database_config_value_date)){
+            exit 1, "Missing date created value for locidex database entry: ${db_name}"
+        }
+        def date_value = value[params.locidex.manifest_config_key][params.locidex.database_config_value_date]
+        def date_check = new SimpleDateFormat(params.locidex.date_format_string).parse(date_value)
+        dates << date_check
+        if(date_check > max_date){
+            max_date = date_check
+            max_date_idx = idx
+        }
+        idx += 1
+    }
+
+    def max_date_count = dates.count(max_date)
+    if(max_date_count > 1){
+        exit 1, "There are multiple versions of the most recent database for ${db_name}. Mikrokondo could not determine the best database to pick."
+    }
+    return db_values[max_date_idx]
+}
+
+
+def tokenize_values(species, match_size){
```

Comment:
Could you clarify what this function is doing? I don't really understand how the sizes are being used when tokenizing.

Reply:
I renamed the function to express what it is actually doing and added an example to the function. It is actullay "windowing" a string.

Reply:
fixed in: https://github.com/phac-nml/mikrokondo/pull/96/commits/dccd27f27d6ce1c4176d13cbe3d08b09fd0b7f2b